src/gui/editors/PianoRoll.cpp

```diff
@@ -4126,6 +4126,15 @@ void PianoRoll::pasteNotes()
 			cur_note.restoreState( list.item( i ).toElement() );
 			cur_note.setPos( cur_note.pos() + Note::quantized( m_timeLine->pos(), quantization() ) );
 
+			if ((cur_note.length().getTicks() == -DefaultTicksPerBar) && (m_pattern->type() != Pattern::BeatPattern)) 
+			{
+				// the length of copied notes is set to the length of the instruments envelope
+				f_cnt_t envFrames = m_pattern->instrumentTrack()->envFrames();
+				// if notes have zero length due to rounding set the length to a small value
+				tick_t noteLength = qMax(static_cast<float>(1), envFrames / Engine::framesPerTick());
```

Comment:
Would `1.0f` work here instead of `static_cast<float>(1)`?

Also, might be good to note that `tick_t` isn't a float/double type, but an integer type.

Reply:
Thanks for the review, maybe something like
`tick_t noteLength = static_cast<tick_t>(qMax(1, round(envFrames / Engine::framesPerTick())))`;
would be better?

Reply:
I was thinking something like this:
```suggestion
				tick_t noteLength = static_cast<tick_t>(qMax(1.0f, envFrames / Engine::framesPerTick()));
```
With the static cast, we drop the remainder completely. Thinking about this though, we may always want to round up otherwise the note would be cut off from a complete playthrough.
If we always want to round up:
```suggestion
				tick_t noteLength = static_cast<tick_t>(qMax(1.0f, ceil(envFrames / Engine::framesPerTick())));
```